### PR TITLE
Workflow Executions: Update Samples Option

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Adds in `update_samples` boolean to `WorkflowExecution` model, which when set to true will update the samples with metadata and files produced by the pipeline.

Updates NextflowComponent to include a `allowed_to_update_samples` boolean which will conditionally display the `update_samples` checkbox.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

As a User with `maintainer` or `owner` access for all of the samples

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as your preferred user
2. Go to Projects Dashboard
3. Click on a Project where you have `analyst` role
4. Navigate to Samples page for the Project
5. Select all samples and launch a pipeline
6. Verify that `update_samples` checkbox is not present and that when executing with sapporo the pipeline completes successfully and does not get stuck in `completing` state, and that files and metadata are not written to the Samples
7. Go to Projects Dashboard
8. Click on a Project where you have `owner` or `maintainer` role
9. Navigate to Samples page for the Project
10. Select all samples and launch a pipeline
11. Verify that `update_samples` checkbox is present and then select it, then when executing with sapporo the pipeline completes successfully and does not get stuck in `completing` state. And then verify that samples and metadata are written to the Samples.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by JeffreyThiessen — DISMISSED** [review]
Looks great. Runs are completing in both cases. :+1:

File: config/locales/en.yml [hunk 1]
```diff
@@ -324,6 +324,7 @@ en:
       information_icon: Information icon
     nextflow:
       required: Required
+      update_samples: Update Samples with Analysis Results
```
Comment (joshsadam) [inline config/locales/en.yml:327]: **nitpick:** Can you sentence case this, all the rest of the labels on pipelines are that way.

Comment (ericenns) [reply config/locales/en.yml]: Updated in [dc9b1d1](https://github.com/phac-nml/irida-next/pull/544/commits/dc9b1d19d7440b68ff6cdcea281fe9c173fb4ab6)

**Review by joshsadam — APPROVED** [review]
This works as tested.

**Pull request merged**